src/coreclr/tests/src/Loader/binding/tracing/Helpers.cs

```diff
@@ -0,0 +1,161 @@
+            Assert.AreEqual(expected.RequestingAssemblyLoadContext ?? string.Empty, actual.RequestingAssemblyLoadContext, $"Unexpected value for {nameof(BindOperation.RequestingAssemblyLoadContext)} on event");
+            ValidateAssemblyName(expected.RequestingAssembly, actual.RequestingAssembly, nameof(BindOperation.RequestingAssembly));
+
+            Assert.AreEqual(expected.Success, actual.Success, $"Unexpected value for {nameof(BindOperation.Success)} on event");
+            Assert.AreEqual(expected.ResultAssemblyPath ?? string.Empty, actual.ResultAssemblyPath, $"Unexpected value for {nameof(BindOperation.ResultAssemblyPath)} on event");
+            Assert.AreEqual(expected.Cached, actual.Cached, $"Unexpected value for {nameof(BindOperation.Cached)} on event");
+            ValidateAssemblyName(expected.ResultAssemblyName, actual.ResultAssemblyName, nameof(BindOperation.ResultAssemblyName));
+
+            ValidateHandlerInvocations(expected.AssemblyLoadContextResolvingHandlers, actual.AssemblyLoadContextResolvingHandlers, "AssemblyLoadContextResolving");
+            ValidateHandlerInvocations(expected.AppDomainAssemblyResolveHandlers, actual.AppDomainAssemblyResolveHandlers, "AppDomainAssemblyResolve");
+            ValidateLoadFromHandlerInvocation(expected.AssemblyLoadFromHandler, actual.AssemblyLoadFromHandler);
+
+            ValidateProbedPaths(expected.ProbedPaths, actual.ProbedPaths);
+
+            ValidateNestedBinds(expected.NestedBinds, actual.NestedBinds);
+        }
+
+        public static string GetAssemblyInSubdirectoryPath(string assemblyName)
+        {
+            return Path.Combine(GetSubdirectoryPath(), $"{assemblyName}.dll");
+        }
+
+        public static string GetSubdirectoryPath()
+        {
+            string appPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            return Path.Combine(appPath, "DependentAssemblies");
+        }
+
+        public static string GetProbingFilePath(ProbedPath.PathSource pathSource, string assemblyName, string culture, string baseAssemblyDirectory = null)
+        {
+            return GetProbingFilePath(pathSource, assemblyName, false, culture, baseAssemblyDirectory);
+        }
+
+        public static string GetProbingFilePath(ProbedPath.PathSource pathSource, string assemblyName, bool isExe)
+        {
+            return GetProbingFilePath(pathSource, assemblyName, isExe, null, null);
+        }
+
+        private static string GetProbingFilePath(ProbedPath.PathSource pathSource, string assemblyName, bool isExe, string culture, string baseAssemblyDirectory)
+        {
+            string baseDirectory = baseAssemblyDirectory ?? Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            if (!string.IsNullOrEmpty(culture))
+                baseDirectory = Path.Combine(baseDirectory, culture);
+
+            string extension = isExe ? "exe" : "dll";
```

Comment:
This is set but not used. Refactoring remnant?

Reply:
Doh. Yep, it should be used below.